src/client/analytics.ts

```diff
@@ -146,6 +148,7 @@ export class CoveoAnalyticsClient implements AnalyticsClient, VisitorIdProvider
     private afterSendHooks: AnalyticsClientSendEventHook[];
     private eventTypeMapping: {[name: string]: EventTypeConfig};
```

Comment:
I wonder why this doesn't just use [Record<Keys, Type>](https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type).

```suggestion
    private eventTypeMapping: Record<string, EventTypeConfig>;
```